src/bit-systems/link-system.ts

```diff
@@ -0,0 +1,30 @@
+import { defineQuery, enterQuery } from "bitecs";
+import type { HubsWorld } from "../app";
+import { Link } from "../bit-components";
+import { isHubsRoomUrl, isLocalHubsAvatarUrl, isLocalHubsSceneUrl, isLocalHubsUrl } from "../utils/media-url-utils";
+import { LinkType } from "../inflators/link";
+
+const linkQuery = defineQuery([Link]);
+const linkEnterQuery = enterQuery(linkQuery);
+
+export function linkSystem(world: HubsWorld) {
+  linkEnterQuery(world).forEach(async eid => {
```

Comment:
I think we should avoid async/await in systems. We should use yield instead. If using async/await for a certain purpose please add inline comments.